nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MonitorActivity.java

```diff
@@ -248,13 +262,13 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         final boolean copyAttributes = context.getProperty(COPY_ATTRIBUTES).asBoolean();
         final boolean isClusterScope = isClusterScope(context, false);
         final boolean shouldReportOnlyOnPrimary = shouldReportOnlyOnPrimary(isClusterScope, context);
-        final List<FlowFile> flowFiles = session.get(50);
+        final FlowFile flowFile = session.get();
```

Comment:
Hi @pvillard31. I was reviewing this change (I rebased locally, had no issues). The only concern I have is how you changed the code to read one FlowFile at a time instead of batches. Functionally, I agree that this causes no issues, but I wonder if it's the best approach? You could have looped through the FlowFile's to update attributes, etc... when activity is restored.

I know it's been... a while since you wrote this. Can you recall your motivations?